### PR TITLE
FIX: allow ':' to be used in keynames and still format LiveTable

### PR DESCRIPTION
**Pull request by tacaswell**
## Description

Internally hash the keys we use to format the LiveTable rows.  This ensure that the value of the data keys can not corrupt the formatting.  Opted to go with a hash (because this is completely internal) rather than trying to sanitize the input as it seemed simpler.

## Motivation and Context

fixes a bug that @stashlukj  ran into when trying to use ``:' ` as signal names.

## How Has This Been Tested?

Added another test of LiveTable.  Re-arranged the tests a bit.

### CODE REVIEW

**Comment by tacaswell** [comment]
The blank space in the test is actually 😻 that seems to not existing in the font the diffs are rendered in.

**Comment by klauer** [comment]
Python already has unique hashes for strings - how about `str(hash(key))`? Though I suppose that wouldn't be consistent between processes, so perhaps hashlib is better. 🤷‍♂️

**Comment by tacaswell** [comment]
I just did not think of that, using `str(hash(key))` is definitely better.

**Comment by tacaswell** [comment]
Had to use `f'h{str(hash(k))}'` as the key because other was the format string ended up being '{3405904703012941009:d}` which tells python to look for the too-many-ith positional argument to format.

File: bluesky/tests/test_callbacks.py [hunk 1]
```diff
@@ -266,6 +270,30 @@ def test_table_external(RE, hw, db):
 +------------+--------------+----------------+----------------+"""
 
 
+def test_evil_table_names(RE):
+    from ophyd import Signal
+    sigs = [
+        Signal(value=0, name="a:b"),
+        Signal(value=0, name="a,b"),
+        Signal(value=0, name="a'b"),
+        Signal(value=0, name="🐍"),
```
Comment (klauer) [inline bluesky/tests/test_callbacks.py:279]: 🐍

Comment (tacaswell) [reply bluesky/tests/test_callbacks.py:279]: :snake:

**Comment by tacaswell** [comment]
It is failing on the docs build due to warnings from PIMS

**Comment by tacaswell** [comment]
power-cycled in the hope that the docs are fixed on master.

**Comment by klauer** [comment]
Bump @tacaswell @danielballan, can we get this one in? @ZLLentz found that we're running into this with our default config...

**Comment by ZLLentz** [comment]
New users of `bluesky` who want to scan simple `EpicsSignal` instances run into this because the default `name`, if not provided, is just the pvname again, which always has `:` in it at SLAC.

**Comment by danielballan** [comment]
Yes. We're doing another round of patch releases this week.

Re-ran the two failed builds...

**Pull request merged**